remote.go

```diff
@@ -1253,7 +1253,12 @@ func (r *Remote) ListContext(ctx context.Context, o *ListOptions) (rfs []*plumbi
 }
 
 func (r *Remote) List(o *ListOptions) (rfs []*plumbing.Reference, err error) {
-	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	timeout := o.Timeout
```

Comment:
Worth adding a check here, `if o.Timeout < 0` return an error.

Reply:
Done in 11f41a7